src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,117 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Storage {
+
+    private final String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads tasks from the specified filePath
+     *
+     * @return An ArrayList of Tasks loaded from the file.
+     * @throws DukeException throws exception when Tasks cannot be loaded
+     */
+    public ArrayList<Task> loadTasks() throws DukeException {
+        try {
+            ArrayList<Task> res = new ArrayList<>();
+            File taskFile = new File(this.filePath);
+            if (!taskFile.createNewFile()) {
+                Scanner fileReader = new Scanner(taskFile);
+                while (fileReader.hasNextLine()) {
+                    String data = fileReader.nextLine();
+                    res.add(dataToTask(data));
+                }
+            }
+            return res;
+        } catch (IOException | DukeException e) {
+            throw new DukeException("Unable to load tasks from file.");
+        }
+    }
+
+    private static Task dataToTask(String str) throws DukeException {
+        String[] taskArr = str.split(",");
+        String taskType = taskArr[0];
+        boolean taskDone = taskArr[1].equals("1");
```

Comment:
Maybe the boolean name should be changed to isTaskDone to meet the coding standard.

Reply:
ok, thanks for pointing this out! as well as the other comments given